sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -56,6 +56,10 @@ public static SetAnalyzedStatement analyze(SetStatement node, ParameterContext p
                 }
                 settingsApplier.apply(builder, parameterContext.parameters(), assignment.expression());
             } else {
+                Set<String> settingNames = CrateSettings.settingNamesByPrefix(settingsName);
+                if (settingNames.size() != 0) {
+                    throw new IllegalArgumentException(String.format(Locale.ENGLISH, "setting '%s' is being used with SET SESSION / LOCAL", settingsName));
```

Comment:
suggestion to make it more descriptive: GLOBAL cluster setting cannot be used with SET SESSION/LOCAL.